index.php

```diff
@@ -1 +1,87 @@
 
+<?php
+
+class DoublyLinkedListNode {
+    public $data;
+    public $next;
+    public $prev;
+
+    public function __construct($node_data)
+    {
+        $this->data = $node_data;
+        $this->next = null;
+        $this->prev = null;
+    }
+}
+
+class DoublyLinkedList {
+    public $head;
+    public $tail;
+
+    public function __construct()
+    {
+        $this->head = null;
+        $this->tail = null;
+    }
+
+    public function insertNode($node_data)
+    {
+        $node = new DoublyLinkedListNode($node_data);
+
+        if (is_null($this->head)) {
+            $this->head = $node;
+        } else {
+            $this->tail->next = $node;
+            $node->prev = $this->tail;
+        }
+
+        $this->tail = $node;
+    }
+}
+
+function printDoublyLinkedList($node, $sep, $fptr)
+{
+    while (!is_null($node)) {
+        fwrite($fptr, $node->data);
+
+
+        $node = $node->next;
+
+        if (!is_null($node)) {
+            fwrite($fptr, $sep);
+        }
+    }
+}
+
+// Complete the reverse function below.
+
+/*
+ * For your reference:
+ *
+ * DoublyLinkedListNode {
+ *     int data;
+ *     DoublyLinkedListNode next;
+ *     DoublyLinkedListNode prev;
+ * }
+ *
+ */
+function reverse($head) {
+$newDLL = new DoublyLinkedList();
+        while ($head){
```

Comment:
There is no need to create this (second) DoublyLinkedList: the first list was created while reading from file.
Here you need just to reverse it, what is done starting from line 75